modules/apps/transfer/keeper/relay_test.go

```diff
@@ -261,17 +406,102 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestOnRecvPacketDoesNotSetTotalEscrowAmountForSourceIBCToken() {
+	/*
+		Given the following flow of tokens:
+
+		chain A (channel 0) -> (channel-0) chain B (channel-1) -> (channel-1) chain A (channel-1)             -> (channel-1) chain B
```

Comment:
wouldn't this be channel 0 because chainA only only established ch-0 to transfer to chainB? 

```suggestion
		chain A (channel 0) -> (channel-0) chain B (channel-1) -> (channel-1) chain A (channel-1)             -> (channel-0) chain B
```

Reply:
The test establishes 2 channels between chain A and B. If we send the tokens over channel-0 the receiving chain would actually mint tokens with denom `transfer/channel-0/transfer/channel-1/transfer-channel-0/stake`.

Reply:
hmm still a bit confused

so we have 2 channels from ChainA -> ChainB 
```
channel-0:  ChainA -> ChainB
channel-1:  ChainA -> ChainB

chainA (sends stake denom over ch-0)  -----> ChainB (receives transfer/ch-0/stake)
chainB (sends transfer/ch-0/stake denom over ch-1 ----> ChainA (receives transfer/ch-1/transfer/ch-0/stake) 
```

My question is wouldn't ChainB just send stake as (transfer/ch-0/stake) to chainA Why does it have to send it over a different channel ch-1 instead of ch-1?

Reply:
If we send `transfer/channel-0/stake` from chain B to chain A over `channel-1` then chain A will not be the source chain (`ReceiverChainIsSource` will return false). We need to send `transfer/channel-1/transfer/channel-0/stake` from chain A to chain B over `channel-1`, so that chain B is the source chain for tokens with denom `transfer/channel-0/stake` and `ReceiverChainIsSource` returns true, so that we can test the scenario of receiving IBC denom tokens for which the receiver chain is source, but no total escrow amount is stored in state.

I hope this makes sense now. I know it's complicated sometimes to get one's head around with all these :) so please keep asking questions if something is not clear. :)

Reply:
aah so the whole flow is gonna looks something like this; 

```
ChainA (ch-0) -----> (ch-1) ChainB 
Source = chainA 
ChainA sends [10stake from transfer/ch-0 to transfer/ch-1] to chainB

ChainB (ch-1) ------> (ch-0) ChainA
Source = chainA
ChainB sends [10stake from transfer/ch-1/stake to transfer/ch-0] to chainA
```

my question here is, wouldn't chainA be the source when we make transfer from  chainB --> chainA?  Is chainB sending tokens from a different channel than the one it received it from, therefore chainA is not the source chain here?

Reply:
i think my assumption about sending tokens over different channel than it was received and therefore the chain will be native is right after reading this description you posted (https://github.com/cosmos/ibc-go/pull/3019#discussion_r1115342675)